### ISSUE TITLE
Update root component to React functional component.

### ISSUE THREAD

**Issue by divyanshu-patil** [RocketChat/Rocket.Chat.ReactNative#6031]
### Description:

The root component is currently implemented as a React class component. It should be updated to use a more modern functional component for consistency and to leverage hooks.

```javascript
export default class Root extends React.Component<{}, IState> {
	private listenerTimeout!: any;
	private dimensionsListener?: EmitterSubscription

	constructor(props: any) {
		super(props);
		this.init();
		if (!isFDroidBuild) {
			this.initCrashReport();
		}
		const { width, height, scale, fontScale } = Dimensions.get('window');
		const theme = initialTheme();
		this.state = {
			theme: getTheme(theme),
			themePreferences: theme,
			width,
			height,
			scale,
			fontScale
		};
		if (isTablet) {
			this.initTablet();
		}
		setNativeTheme(theme);
	}
```

### Additional context:

In the future, all components using class components should migrate to functional components to align with modern React practices.

**Comment by diegolmello** [comment]
Can you do it?

**Comment by divyanshu-patil** [comment]
assign it to me

**Comment by Rohit3523** [comment]
Shouldn't we follow the same component structure throughout the entire app, instead of having some as functional components and others as class components?

**Comment by divyanshu-patil** [comment]
> Shouldn't we follow the same component structure throughout the entire app, instead of having some as functional components and others as class components?

Yes, it's best to stick to one approach for consistency. Functional components are simpler, perform better, and work well with hooks, making them the preferred choice. While we can migrate gradually, moving all components to functional ones will make the app easier to maintain and align with current React best practices.
